src/main/java/io/github/Alathra/AlathranCommands/commands/CommandStop.java

```diff
@@ -118,35 +123,72 @@ private static void scheduleStart() {
         isActionScheduled = true;
         actionTaskId = Bukkit.getScheduler().scheduleSyncRepeatingTask(AlathranCommands.getInstance(), () -> {
             final Instant currentTime = Instant.now();
-            final Duration remainingTime = Duration.between(currentTime, actionBeginTime);
+            final Duration remainingTime = Duration.between(currentTime, actionEndTime).abs();
             final boolean shouldStop = currentTime.isAfter(actionEndTime) && !Bukkit.isStopping();
 
             if (shouldStop) {
-                Bukkit.broadcast(ColorParser.of("&a[&cScheduler&a] Stopping the server...").build());
-                Bukkit.shutdown();
-                return;
-            }
-
-            // Announce minutes remaining
-            if (remainingTime.toSecondsPart() == 0) {
-                switch (Math.toIntExact(remainingTime.toMinutes())) {
-                    case 30, 10, 5, 4, 3, 2, 1 -> {
-                        Bukkit.broadcast(ColorParser.of("&a[&cScheduler&a] Stopping the server in <time> minutes.").build());
+                switch (actionType) {
+                    case STOP -> {
+                        Bukkit.broadcast(ColorParser.of(MiscCfg.get().getString("Scheduler.Messages.Stop-now")).parseMinimessagePlaceholder("scheduler", MiscCfg.get().getString("Scheduler.Scheduler-tag")).build());
+                        Bukkit.shutdown();
+                        return;
+                    }
+                    case RESTART -> {
+                        Bukkit.broadcast(ColorParser.of(MiscCfg.get().getString("Scheduler.Messages.Restart-now")).parseMinimessagePlaceholder("scheduler", MiscCfg.get().getString("Scheduler.Scheduler-tag")).build());
+                        Bukkit.spigot().restart();
                         return;
                     }
                 }
+
             }
 
-            // Announce seconds remaining
-            if (remainingTime.toMinutes() == 0) {
-                switch (Math.toIntExact(remainingTime.toSeconds())) {
-                    case 30, 20, 10, 5, 4, 3, 2, 1 -> {
-                        Bukkit.broadcast(ColorParser.of("&a[&cScheduler&a] Stopping the server in <time> seconds.").build());
-                        return;
+            switch (actionType) {
+                case STOP -> {
```

Comment:
These massive `cases` are duplicated code other than the config strings.

It would be a lot less code if you remove the `switch` on line 145, and instead put the `Bukkit#broadcast` methods in a switch case.